api/src/main/java/net/starype/quiz/api/game/answer/DoubleCorrectnessEvaluator.java

```diff
@@ -0,0 +1,30 @@
+package net.starype.quiz.api.game.answer;
+
+import java.util.Set;
+
+public class DoubleCorrectnessEvaluator implements CorrectnessEvaluator {
+
+    private final Set<Answer> acceptedAnswers;
+    private final double acceptedRange;
+    private final LossFunction lossFunction;
+
+    public DoubleCorrectnessEvaluator(Set<Answer> acceptedAnswers, double acceptedRange, LossFunction lossFunction) {
+        this.acceptedAnswers = acceptedAnswers;
+        this.acceptedRange = Math.max(Math.abs(acceptedRange), 0.0000001);
+        this.lossFunction = lossFunction;
+    }
+
+    @Override
+    public double getCorrectness(Answer answer) {
+        double proposedAnswer = answer.asDouble();
+        double closeness = acceptedAnswers.stream()
+                .map(accepted -> Math.abs(proposedAnswer - accepted.asDouble()))
+                .filter(n -> n < acceptedRange)
+                .min(Double::compareTo)
+                .orElse(acceptedRange + 1.0);
+        if(closeness >= acceptedRange + 1.0)
+            return 0.0;
+        return lossFunction.evaluate(closeness, 0.0, acceptedRange);
+    }
```

Comment:
Here there is a big repetition with the IntCorrectnessEvaluator, it should be possible to remove it

Reply:
I don't thing it's as this using double and int use int. I don't wanted to implement generic type in java